tools/rg_generate_shaders_gl.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+"""
+Converts a fragment or vertex shader file into a GL shader file.
+
+You can use it like this:
+
+    $ ./rg_generate_shaders_gl.py --cmake-output \\
+        ../src/rendergraph/shaders/generated_shaders_gl.cmake \\
+        ../src/rendergraph/shaders/*.{vert,frag}
+"""
+import argparse
+import logging
+import os
+import pathlib
+import re
+import shutil
+import subprocess
+import tempfile
+import typing
+
+
+def find_executable(
+    executable_name: str, additional_paths: list[str] | None = None
+) -> pathlib.Path:
+    """Find an executable by name in $PATH and in the additional paths."""
+    if executable_path := shutil.which(executable_name):
+        return pathlib.Path(executable_path)
+
+    if additional_paths:
+        if executable_path := shutil.which(
+            executable_name, path=os.pathsep.join(additional_paths)
+        ):
+            return pathlib.Path(executable_path)
+
+    raise OSError(f"Executable {executable_name!r} not found!")
+
+
+QSB_EXECUTABLE = find_executable(
+    "qsb",
+    additional_paths=[
+        "/usr/lib/qt6/bin",
+        "/lib/qt6/bin",
+        "/usr/local/lib/qt6/bin",
+    ],
```

Comment:
Should we use `$PATH` as well?
```suggestion
    additional_paths=[
        "/usr/lib/qt6/bin",
        "/lib/qt6/bin",
        "/usr/local/lib/qt6/bin",
    ] + [os.getenv("PATH", "").split(":")],
```

Reply:
We call `shutil.which` up totwo times in the funtion, the first time without setting the path argument which implicitly default to `$PATH`.

Reply:
Ah good to know. Thanks for the lesson!